README.Rmd

```diff
@@ -46,28 +46,113 @@ forcats is now part of the core tidyverse, so you do not need to load it explici
+
+```{r fct_infreq_skin}
+ggplot(starwars, aes(x = fct_infreq(skin_color))) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+Here we have a lot of different skin colors - let's reduce it to only be the top 5. We can use `fct_lump` to "lump" all the infrequent colors into one factor, "other." The argument `n` is the number of levels we want to keep.
+
+```{r fct_lump}
+ggplot(starwars, aes(x = fct_lump(skin_color, n = 5))) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+We could also have used `prop` instead, which keeps all the levels that appear at least `prop` of the time. For example, let's keep skin colors that at least 10% of the characters have: 
+
+```{r fct_lump_prop}
+ggplot(starwars, aes(x = fct_lump(skin_color, prop = .1))) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+Only light and fair remain; everything else is other. 
+
+What if we wanted to see if the average mass differed by eye color? We'll only look at the 6 most popular eye colors and remove NAs.  
+
+```{r fct_lump_mean}
+starwars %>%
+  mutate(eye_color = fct_lump(eye_color, n = 6)) %>%
+  group_by(eye_color) %>%
+  summarise(mean_mass = mean(mass, na.rm = TRUE)) %>%
+  ggplot(aes(x = eye_color, y = mean_mass)) + 
+  geom_col()
+```
+
+It looks like people (or at least one person) with orange eyes is definitely heavier! This graph would look nice though if it was ordered by `mean_mass`. We can do this with `fct_reorder`, which reorders one variable by another. 
+
+```{r fct_reorder}
+starwars %>%
+  mutate(eye_color = fct_lump(eye_color, n = 6)) %>%
+  group_by(eye_color) %>%
+  summarise(mean_mass = mean(mass, na.rm = TRUE)) %>%
+  mutate(eye_color = fct_reorder(eye_color, mean_mass)) %>%
+  ggplot(aes(x = eye_color, y = mean_mass)) + 
+  geom_col()
+```
+
+Let's switch to using another dataset, `gss_cat`, the general social survey. What is the income distribution among the respondents? 
+
+```{r initial_plot2}
+ggplot(gss_cat, aes(x = rincome)) +
+   geom_bar() + 
+   coord_flip()
+```
+
+Notice that the income levels are in the correct order - they go from `"Lt $1000"` to `$25,000 or more`. This is not a coicidence. When you're working with ordinal data, where there is an order, you can have an ordered factor. You can examine them with the base function `levels`, which prints them in order:
 
 ```{r}
-x1 <- c("Dec", "Apr", "Jan", "Mar")
-month_levels <- c(
-  "Jan", "Feb", "Mar", "Apr", "May", "Jun", 
-  "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
-)
+levels(gss_cat$rincome)
+```
 
-factor(x1, month_levels)
+But what if your factor came in the wrong order? Let's simulate that by reordering the levels of `rincome` randomly with `fct_shuffle` and then plotting it again.
 
-parse_factor(x1, month_levels)
+```{r fct_shuffle}
+ggplot(gss_cat, aes(x = fct_shuffle(rincome))) +
+   geom_bar() + 
+   coord_flip()
 ```
 
-The advantage of `parse_factor()` is that it will generate a warning if values of `x` are not valid levels:
+How can we fix this and put it in the right order?
```

Comment:
Maybe have a subheader in here about reordering for navigability?

Reply:
That sounds helpful. What do you think of adding a couple headers, so we don't have the large "getting started" chunk? Could do: 

Ordering by frequency (instead of getting started)

Combining levels
 
Combining by another variable

Manually reordering